### PR TITLE
[ENH] Apply derivative definition to qMRI appendix

### LINKED ISSUE #1812: update qMRI to align with the new definition of derivatives
1. qmri appendix defines quantitative maps as derivatives regardless of how they were generated

   https://bids-specification.readthedocs.io/en/latest/appendices/qmri.html#quantitative-maps-are-derivatives

   > Regardless of how they are obtained (pre- or post-generated), qMRI maps are stored in the derivatives directory.

   One exception to this in the qMRI page is the recommendation to store scanner-generated UNIT1 files along with the raw data.

2. The schema files rules for quantitative maps under the rules for raw datasets.

   https://github.com/bids-standard/bids-specification/blob/0950f6df9e9c6d598fcc50e269f350066d78b7a0/src/schema/rules/files/raw/anat.yaml#L34

3. The legacy valiator (version 1.14.5) is completely fine with having those quantitative maps in a raw dataset (tested on several of the bids examples).

---

- **1** contradicts the operational definition of derivatives datasets that was agreed on in Copenhagen last year.

  https://bids-specification.readthedocs.io/en/latest/derivatives/introduction.html#bids-derivatives

  > Raw data are data that have been curated into BIDS from a non-BIDS source. If a dataset is derived from at least one other valid BIDS dataset, then it is a derivative dataset.

- **1** contradicts the behavior of the legacy validator (**3"") and, **1** and **2** tends to contradict each other as the schema seems to imply that quantitative maps are legit raw data (I would expect that the deno validator would behave the same as the legacy one given the schema).

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes #1812
---

- update qMRI sections that mention qmaps: https://bids-specification--1813.org.readthedocs.build/en/1813/appendices/qmri.html#outputs-are-quantitative-maps

### CODE REVIEW

File: src/appendices/qmri.md [hunk 1]
```diff
@@ -133,10 +133,13 @@ A guide for using macros can be found at
 
 Please visit the [file collections appendix](./file-collections.md#magnetic-resonance-imaging) to see the list of currently supported qMRI applications.
 
-### Quantitative maps are derivatives
+### Outputs are quantitative maps
 
-Regardless of how they are obtained (pre- or post-generated), qMRI maps are stored in the `derivatives` directory.
-For example a `T1map` can be generated from an `MP2RAGE` file collection using either options.
+qMRI maps are stored differently depending on the process that generated them.
+Pre-generated qMRI maps MUST be stored as part of raw BIDS datasets,
```
Comment (effigies) [inline src/appendices/qmri.md:139]:
```suggestion
Pre-generated qMRI maps MAY be stored as part of raw BIDS datasets,
```

This is unenforceable and I don't think we should require it, anyway.

Somebody might prefer to create a dataset like:

```
derivatives/
  scanner-generated/
    sub-01/
      anat/
        sub-01_T1map.nii.gz
        sub-01_T1map.json
        sub-01_UNIT1.nii.gz
        sub-01_UNIT1.json
  open-algorithm/
    sub-01/
      anat/
        sub-01_T1map.nii.gz
        sub-01_T1map.json
        sub-01_UNIT1.nii.gz
        sub-01_UNIT1.json
sub-01/
  anat/
    sub-01_inv-1_part-mag_MP2RAGE.nii.gz
    sub-01_inv-1_part-phase_MP2RAGE.nii.gz
    sub-01_inv-1_MP2RAGE.json
    sub-01_inv-2_part-mag_MP2RAGE.nii.gz
    sub-01_inv-2_part-phase_MP2RAGE.nii.gz
    sub-01_inv-2_MP2RAGE.json
```

There the symmetry between the different derivatives would make the dataset more legible.

File: src/appendices/qmri.md [hunk 2]
```diff
@@ -526,12 +544,13 @@ as an input to offline calculation of a `T1map` using a dictionary lookup approa
...
-for quantifying a `T1map`.
+Therefore, the `UNIT1` image provided by the scanner
+MUST be stored under the `anat` in a raw BIDS dataset
+along with the `MP2RAGE` file collection
+and to be used as the primary input for quantifying a `T1map`.
 
-If an additional `UNIT1` image is calculated offline, then the output is to be stored in the
-`derivatives` directory with necessary provenance information.
+If an additional `UNIT1` image is calculated offline,
+then the output is to be stored in a derivative BIDS dataset with necessary provenance information.
```
Comment (Remi-Gau) [inline src/appendices/qmri.md:553]:
```suggestion
then the output MAY be stored in a derivative BIDS dataset with necessary provenance information.
```

@effigies here too?

File: src/schema/objects/common_principles.yaml [hunk 3]
```diff
@@ -46,6 +46,9 @@ dataset:
   description: |
     A set of neuroimaging and behavioral data acquired for a purpose of a particular study.
     A dataset consists of data acquired from one or more subjects, possibly from multiple sessions.
+derivative:
```
Comment (Remi-Gau) [inline src/schema/objects/common_principles.yaml:49]: wonder if this should be not part of a separate PR

File: src/appendices/qmri.md [hunk 4]
```diff
@@ -136,8 +136,8 @@ Please visit the [file collections appendix](./file-collections.md#magnetic-reso
 ### Outputs are quantitative maps
 
 qMRI maps are stored differently depending on the process that generated them.
-Pre-generated qMRI maps MUST be stored as part of raw BIDS datasets,
-where as it MUST be stored in a derivative BIDS dataset if it were post-generated.
+Pre-generated qMRI maps MAY be stored as part of a raw BIDS dataset,
+where as it MAY be stored in a derivative BIDS dataset if it were post-generated.
```
Comment (effigies) [inline src/appendices/qmri.md:140]: I think this one should be a MUST, even though it's unenforceable. Here the MUST would be best interpreted as a directive to a tool that does the calculation that it should not modify the original dataset.

```suggestion
where as they MUST be stored in a derivative BIDS dataset if they were post-generated.
```

Also, updating pronouns for agreement with "maps".

File: src/appendices/qmri.md [hunk 5]
```diff
@@ -526,12 +544,13 @@ as an input to offline calculation of a `T1map` using a dictionary lookup approa
 provided by the stock sequence. Instead, the `magnitude` and `phase` images are exported. Please
 see the relevant discussion at [qMRLab issue #255](https://github.com/qMRLab/qMRLab/issues/255).
 
-Therefore, the `UNIT1` image provided by the scanner is RECOMMENDED to be stored under the `anat`
-raw dataset directory along with the `MP2RAGE` file collection and to be used as the primary input
-for quantifying a `T1map`.
+Therefore, the `UNIT1` image provided by the scanner
+MUST be stored under the `anat` in a raw BIDS dataset
```
Comment (effigies) [inline src/appendices/qmri.md:548]:
```suggestion
SHOULD be stored under the `anat` in a raw BIDS dataset
```

File: src/appendices/qmri.md (hunk 2 above)
Comment (effigies) [reply src/appendices/qmri.md:553]: Here I would say MUST:

```suggestion
then the output MUST be stored in a derivative BIDS dataset with necessary provenance information.
```

File: src/schema/objects/common_principles.yaml (hunk 3 above)
Comment (effigies) [reply src/schema/objects/common_principles.yaml:49]: Yeah. It's a bit hidden from potential commenters, here.

Comment (Remi-Gau) [reply src/schema/objects/common_principles.yaml:49]: OK will move this out
better change log in the end too

**Comment by Remi-Gau** [comment]
Will create a companion PR to modify a couple of bids example to make that the validator has got something to sink its teeth into.

**Review by effigies — APPROVED** [review]

**Comment by Remi-Gau** [comment]
@agahkarakuzu can you maybe cross-check what I did here?

File: src/appendices/qmri.md [hunk 6]
```diff
@@ -133,10 +133,13 @@ A guide for using macros can be found at
...
-### Quantitative maps are derivatives
+### Outputs are quantitative maps
 
-Regardless of how they are obtained (pre- or post-generated), qMRI maps are stored in the `derivatives` directory.
-For example a `T1map` can be generated from an `MP2RAGE` file collection using either options.
+qMRI maps are stored differently depending on the process that generated them.
+Pre-generated qMRI maps MAY be stored as part of a raw BIDS dataset,
+where as they MUST be stored in a derivative BIDS dataset if they were post-generated.
+
+See the example below of a `T1map` generated from an `MP2RAGE` file collection using either options.
```
Comment (tsalo) [inline src/appendices/qmri.md:142]:
```suggestion
See the example below of a `T1map` generated from an `MP2RAGE` file collection using either option.
```

File: src/appendices/qmri.md [hunk 7]
```diff
@@ -133,10 +133,13 @@ A guide for using macros can be found at
...
 Please visit the [file collections appendix](./file-collections.md#magnetic-resonance-imaging) to see the list of currently supported qMRI applications.
 
-### Quantitative maps are derivatives
+### Outputs are quantitative maps
 
-Regardless of how they are obtained (pre- or post-generated), qMRI maps are stored in the `derivatives` directory.
-For example a `T1map` can be generated from an `MP2RAGE` file collection using either options.
+qMRI maps are stored differently depending on the process that generated them.
+Pre-generated qMRI maps MAY be stored as part of a raw BIDS dataset,
+where as they MUST be stored in a derivative BIDS dataset if they were post-generated.
```
Comment (tsalo) [inline src/appendices/qmri.md:140]:
```suggestion
whereas they MUST be stored in a derivative BIDS dataset if they were post-generated.
```

**Comment by agahkarakuzu** [comment]
@Remi-Gau the changes you introduced looks good to me! This was initially what we wanted to do, but at that time it was not really clear. Thank you!
